common/G4_hFarBwdBeamLine_EIC.C

```diff
@@ -0,0 +1,313 @@
+  Enable::HFARBWD_MAGNETS_IP8 |= Enable::HFARBWD_MAGNETS and Enable::IP8;
+
+  Enable::HFARBWD_VIRTUAL_DETECTORS_IP6 |= Enable::HFARBWD_VIRTUAL_DETECTORS and Enable::IP6;
+  Enable::HFARBWD_VIRTUAL_DETECTORS_IP8 |= Enable::HFARBWD_VIRTUAL_DETECTORS and Enable::IP8;
+
+  if (Enable::HFARBWD_MAGNETS_IP6 && Enable::HFARBWD_MAGNETS_IP8)
+  {
+    cout << "You cannot have magnets for both IP6 and IP8 ON at the same time" << endl;
+    gSystem->Exit(1);
+  }
+
+  if (Enable::HFARBWD_MAGNETS_IP6)
+  {
+//    hFarBwdBeamLine::enclosure_z_max = -4500.;
+    hFarBwdBeamLine::enclosure_z_max = -2000.;
+    BlackHoleGeometry::min_z = std::min(BlackHoleGeometry::min_z, hFarBwdBeamLine::starting_z);
+    hFarBwdBeamLine::enclosure_r_max = 200.;
+  }
+
+  if (Enable::HFARBWD_MAGNETS_IP8)
+  {
+//    hFarBwdBeamLine::enclosure_z_max = -4500.;
+    hFarBwdBeamLine::enclosure_z_max = -2000.;
+    BlackHoleGeometry::min_z = std::min(BlackHoleGeometry::min_z, hFarBwdBeamLine::starting_z);
+    hFarBwdBeamLine::enclosure_r_max = 200.;
+  }
+
+  hFarBwdBeamLine::enclosure_center = 0.5 * (hFarBwdBeamLine::starting_z + hFarBwdBeamLine::enclosure_z_max);
+
+  BlackHoleGeometry::max_z = std::max(BlackHoleGeometry::max_z, hFarBwdBeamLine::enclosure_z_max);
+  BlackHoleGeometry::max_radius = std::max(BlackHoleGeometry::max_radius, hFarBwdBeamLine::enclosure_r_max);
+}
+
+void hFarBwdDefineMagnets(PHG4Reco *g4Reco)
+{
+
+//  cout << " HERE ???? " << endl;
+//  gSystem->Exit(1);
+//  exit(0);
+
+  bool overlapCheck = Enable::OVERLAPCHECK || Enable::HFARBWD_OVERLAPCHECK;
+  int verbosity = std::max(Enable::VERBOSITY, Enable::HFARBWD_VERBOSITY);
+
+  hFarBwdBeamLine::hFarBwdBeamLineEnclosure = new PHG4CylinderSubsystem("hFarBwdBeamLineEnclosure");
+//  hFarBwdBeamLine::hFarBwdBeamLineEnclosure->set_double_param("place_z", -2450);
+  hFarBwdBeamLine::hFarBwdBeamLineEnclosure->set_double_param("place_z", hFarBwdBeamLine::enclosure_center);
+  hFarBwdBeamLine::hFarBwdBeamLineEnclosure->set_double_param("radius", 0);
+  hFarBwdBeamLine::hFarBwdBeamLineEnclosure->set_double_param("thickness", hFarBwdBeamLine::enclosure_r_max);  // This is intentionally made large 25cm radius
+//  hFarBwdBeamLine::hFarBwdBeamLineEnclosure->set_double_param("length", 4000);
+  hFarBwdBeamLine::hFarBwdBeamLineEnclosure->set_double_param("length", fabs(hFarBwdBeamLine::enclosure_z_max - hFarBwdBeamLine::starting_z));
+  hFarBwdBeamLine::hFarBwdBeamLineEnclosure->set_string_param("material", "G4_Galactic");
+  hFarBwdBeamLine::hFarBwdBeamLineEnclosure->set_color(.5, .5, .5, 0.2);
+  hFarBwdBeamLine::hFarBwdBeamLineEnclosure->OverlapCheck(overlapCheck);
+  if (verbosity)
+    hFarBwdBeamLine::hFarBwdBeamLineEnclosure->Verbosity(verbosity);
+  g4Reco->registerSubsystem(hFarBwdBeamLine::hFarBwdBeamLineEnclosure);
+
+  string magFile;
+  if (Enable::HFARFWD_MAGNETS_IP6)
+    magFile = string(getenv("CALIBRATIONROOT")) + "/Beam/ip6_h_farFwdBeamLineMagnets.dat";
```

Comment:
> I have trouble seeing the far backward detectors, please double check the inline comments:
> 
> ![image](https://user-images.githubusercontent.com/7947083/124693905-37656600-deae-11eb-8900-af233a73d2c9.png)

Hi Jin, I can confirm that we are reading the correct data file, the far forward and the far backward data file are the same files.

Reply:
Great. 

As in the above display, do you also see the far backward electron travel much further than the vacuum enclosure and start air showering much down in z? This would be just an overhead to production without physical gain. 

Would you like to try move the blackhole min-z to the end of the vacuum enclosure?  I have trouble to find out who set the blackhole  min-z to be such a large negative number

Reply:
Hi Jin, 
Sorry that I missed this message, I will attempt your suggestion now.

Bill

Reply:
Hi Jin, the newest update should address this issue. The z_min for the particle is set to -46.3m, I set the enclosure to now end at -47m. We should be good to go : )

Reply:
Double-check  
![Screenshot from 2021-07-07 18-58-32](https://user-images.githubusercontent.com/5608342/124839041-06d90700-df56-11eb-9366-f5d0cf40ea49.png)